src/Duration.ts

```diff
@@ -0,0 +1,54 @@
+var unitValues = [1, 1000, 60000, 3600000, 86400000];
+
+function convertUnits(time: number, unitMilliseconds: number) {
+	return (time > 0 ? 1 : -1) * Math.floor(Math.abs(time) / unitMilliseconds);
+}
+
+export default class Duration {
+	public static SECOND = unitValues[1];
+	public static MINUTE = unitValues[2];
+	public static HOUR = unitValues[3];
+	public static DAY = unitValues[4];
+
+	public time: number = 0;
+
+	constructor(... rest: number[]) {
+		var time = 0;
+		for (var i = 0; i < rest.length; i++) {
+			time += rest[i] * unitValues[i];
+		}
+		this.time = time;
+	}
+
+	get milliseconds(): number {
+		return this.time % Duration.SECOND;
```

Comment:
I talked with @devpaul and the mod is to make the `duration.milliseconds` API similar to `dateObject.milliseconds` which makes perfect sense.